explainability-service/src/main/java/org/kie/trustyai/service/prometheus/PrometheusPublisher.java

```diff
@@ -38,7 +41,14 @@ public PrometheusPublisher(MeterRegistry registry) {
     private void createOrUpdateGauge(String name, Iterable<Tag> tags, UUID id) {
         Gauge.builder(name, new AtomicDouble(), value -> values.get(id).doubleValue())
                 .tags(tags).strongReference(true).register(registry);
+    }
 
+    public void removeGauge(String name, UUID id) {
+        Search s = this.registry.find(METRIC_PREFIX + name.toLowerCase());
+        Collection<Gauge> gaugesToDelete = s.tags(List.of(Tag.of("request", id.toString()))).gauges();
+        for (Gauge g : gaugesToDelete) {
+            registry.remove(g);
```

Comment:
Out of curiosity, how does this work? Does it remove the metric from the registry so it is not published anymore but old values remain in Prometheus or this is an actual remove that cleanup Prometheus storage?
I think the behaviour we want is just to remove new _empty_ values but to keep old values, am I right?

Reply:
So this does not clean up Prometheus, but merely our publisher. So old values published to Prometheus will persist. This unpublishes the deleted metric request from the TrustyAI /q/metrics endpoint, as otherwise a line like:
[previously deleted metric]: 0.xxx
would permanently remain in /q/metrics and therefore be scraped by Prometheus as an active metric, i.e, with a new timestamp every scrape, and so you'd see all deleted metrics as infinite flat lines in Prometheus